### ISSUE TITLE
Smarty substitutions seems to 'eat' random characters

### ISSUE THREAD

**Issue by TheWiseMan** [BenjaminHoegh/ParsedownExtended#61]
### Describe the bug

When using smarty substitutions, characters that should be left untouched disappear from the output.

![image](https://github.com/BenjaminHoegh/ParsedownExtended/assets/45821875/4c09bf24-cd3a-4b02-8328-f4937ad27731)
![image](https://github.com/BenjaminHoegh/ParsedownExtended/assets/45821875/928b7c04-b736-4057-96b9-024c49cf99d1)
```php

    $Parser->setSetting('smarty', [
    'substitutions' => [
        'left-double-quote' => '"', // Double bottom quote
        'right-double-quote' => '"', // Double top quote
        "ndash"=> '–',
        "mdash" =>'—',
        "left-single-quote" => "'",
        "right-single-quote" => "'",
        "left-angle-quote" => '«',
        "right-angle-quote" => '»',
        "ellipses"=> "..."
    ],

    "enabled" => true
    ]);
```

Upon disbling smarty, everything works fine (expect substitutions of course).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by BenjaminHoegh** [comment]
Can I get you to post the text from your image, it makes it a bit easier to debug

**Comment by TheWiseMan** [comment]
Sure, do you want the full document ? (sorry for the delay)

**Comment by TheWiseMan** [comment]
It unfortunately is, but testing with a test file containing solely the line below triggers the bug :

```md
# Linéariser les équations d'évolution de la position $r$ de la bille et de la position angulaire $varphi$ du rail autour de $r=r_0=0.25m$ et $varphi=0$. Montrer qu'on peut les écrire sous le formalisme d'état.
```

![image](https://github.com/BenjaminHoegh/ParsedownExtended/assets/45821875/8735126b-717d-432a-a794-a1a20e1097ba)

**Comment by BenjaminHoegh** [comment]
Fixed in release [1.2.7](https://github.com/BenjaminHoegh/ParsedownExtended/releases/tag/v1.2.7)
